pocs/tests/test_pocs.py

```diff
@@ -192,13 +191,32 @@ def test_is_weather_safe_no_simulator(pocs):
     assert pocs.is_weather_safe() is False
 
 
+def wait_for_message(sub, type=None, attr=None, value=None):
+    """Wait for a message of the specified type and contents."""
+    assert (attr is None) == (value is None)
+    while True:
+        msg_type, msg_obj = sub.receive_message()
+        if not msg_obj:
+            continue
+        if type and msg_type != type:
+            continue
+        if not attr or attr not in msg_obj:
+            continue
+        if value and msg_obj[attr] != value:
+            continue
+        return msg_type, msg_obj
+
+
 def test_run_wait_until_safe(observatory):
     os.environ['POCSTIME'] = '2016-09-09 08:00:00'
 
     # Make sure DB is clear for current weather
     observatory.db.clear_current('weather')
 
     def start_pocs():
+        observatory.logger.info('start_pocs ENTER')
+        # Why is this not ALL hardware (i.e. why change from the default
```

Comment:
It's removing `weather` from the list so we can test for bad weather before setting to good, otherwise the weather simulator would always report good. (which is what the whole test is for)